sql/parser/eval.go

```diff
@@ -1214,11 +1434,61 @@ func evalCastExpr(expr *CastExpr) (Datum, error) {
 		}
 		return s, nil
 
-		// TODO(pmattis): unimplemented.
-		// case *DecimalType:
-		// case *DateType:
-		// case *TimestampType:
+	// TODO(pmattis): unimplemented.
+	// case *DecimalType:
+	case *DateType:
+		switch d.(type) {
+		case DString:
+			return ParseDate(d.(DString))
+		}
+	case *TimestampType:
+		switch d.(type) {
+		case DString:
+			return ParseTimestamp(d.(DString))
+		}
+	case *IntervalType:
+		switch d.(type) {
+		case DString:
+			d, err := time.ParseDuration(string(d.(DString)))
```

Comment:
yeah I was thinking that the go style would be us moving to support a new modern style. I feel the postgres style is a lot like cobol!